mithril-signer/src/transactions_importer_with_prune_decorator.rs

```diff
@@ -0,0 +1,130 @@
+use std::sync::Arc;
+
+use async_trait::async_trait;
+
+use mithril_common::entities::{BlockNumber, ImmutableFileNumber};
+use mithril_common::signable_builder::TransactionsImporter;
+use mithril_common::StdResult;
+
+/// Cardano transactions pruner
+#[cfg_attr(test, mockall::automock)]
+#[async_trait]
+pub trait TransactionPruner: Send + Sync {
+    /// Prune the transactions older than the given number of blocks (based on the block range root
+    /// stored).
+    async fn prune(&self, number_of_blocks_to_keep: BlockNumber) -> StdResult<()>;
+}
+
+/// A decorator of [TransactionsImporter] that prunes the transactions older than a given number of
+/// blocks after running the import.
+///
+/// If the number of blocks to keep is not provided, no pruning is performed.
+pub struct TransactionsImporterWithPruneDecorator {
```

Comment:
IMO the fact that it is a decorator is an implementation detail which should not appear in the name:
```suggestion
pub struct TransactionsImporterWithPruner {
```